src/main/java/com/minecolonies/coremod/compatibility/jei/JEIPlugin.java

```diff
@@ -155,6 +145,66 @@ public void onRuntimeAvailable(@NotNull final IJeiRuntime jeiRuntime)
         this.weakRuntime = new WeakReference<>(jeiRuntime);
     }
 
+    private Map<IRecipeType<?>, List<IGenericRecipe>> cacheVanillaRecipes()
```

Comment:
Unless I am missing something, but I think you are missing the cache here.

You loop the recipe manager and build a map with the results.

That map is never stored anywhere for reuse and the map is rebuild every time...

Can you point out to me how this is a caching function?
If so then please document this since the functionality is not obvious on first sight.

Reply:
It needs to be built once each time this method is called because this is called on reload when the input recipes have all changed.  As such there is no point in storing it anywhere.

As was already stated in the description, how this acts as a cache is that this list of generic recipes is then used repeatedly when scanning the individual crafting modules.  Prior to this change it would rebuilt the list from scratch for each and every module.

Reply:
Ahhhh okey.
I would maybe rename this then to buildRecipes.
Since this does not actually do caching, but instead just builds them and then other code reuses the result.